Library/Homebrew/cmd/readall.rb

```diff
@@ -69,4 +74,17 @@ def readall
       end
     end
   end
+
+  def syntax_errors_or_warnings?(rb)
+    # Syntax errors are printed to stderr and result in a non-zero exit code.
+    # Syntax warnings are also printed to stderr, but don't alter the exit code.
+    if Open3.respond_to?(:capture3)
+      _out, err, status = Open3.capture3(RUBY_PATH, "-c", "-w", rb)
+      $stderr.print err
+      !status.success? || !err.chomp.empty?
+    else
+      # Fallback for Ruby 1.8.7. Doesn't flag syntax warnings as failure.
+      system RUBY_PATH, "-c", "-w", rb
```

Comment:
Feels weird to have separate paths here. Could we just use `popen_read` and `$?` to do this on 1.8?